File: C/moveZeroes.c
#include<stdlib.h>
#include<stdio.h>

void moveZero(int arr[], int n) {
    int i = 0;
    int j = 0;

    while(j < n) {
        if(arr[j] == 0)
        j++;
        else {
            arr[i] = arr[j];
            i++;
            j++;
        }
    }
    while(i < n) {
        arr[i] = 0;
        i++;
    }
}

int main() {

    int *A,n;
    printf("Enter length of array\n");
    scanf("%d", &n);

    A = (int*)malloc(sizeof(int)*n);
    printf("Enter element in array\n");
    for(int i = 0; i < n; i++) {
        scanf("%d", &A[i]);
    }

    moveZero(A,n);
    for(int i = 0; i < n; i++) {
        printf("%d\t", A[i]);
    }

    return 0;
}

File: JAVA/moveZeroes.java
package JAVA;
import java.util.*;

public class moveZeroes {

    public static void main(String[] args) {
        Scanner t = new Scanner(System.in);
        System.out.println("Enter length of array");
        int n = t.nextInt();
        int arr[] = new int[n];
        System.out.println("Enter element in array");
        int i = 0;
        while(i < n) {
            arr[i] = t.nextInt();
            i++;
        }

        Move m = new Move();
        m.moveZero(arr);
        int j = 0;
        while(j < n) {
            System.out.print(arr[j]+" ");
            j++;
        }
    }
}

class Move {

    public void moveZero(int [] nums) {
        int i = 0;
        int j = 0;
        while(j < nums.length) {
            if(nums[j] == 0)
            j++;
            else {
                nums[i] = nums[j];
                i++;
                j++;
            }
        }
        while(i < nums.length) {
            nums[i] = 0;
            i++;
        }
    }
}